opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -546,6 +581,40 @@ std::unique_ptr<Matrix> blockJacobiAdjacency(const Grid& grid,
             return weights;
         }
 
+        Vector getTrueImpesWeightsAnalytic(int pressureVarIndex) const
+        {
+            Vector weights(rhs_->size());
+            ElementContext elemCtx(simulator_);
+            Amg::getTrueImpesWeightsAnalytic(pressureVarIndex, weights, simulator_.vanguard().gridView(),
+                                             elemCtx, simulator_.model(),
+                                             ThreadManager::threadId());
+            return weights;
+        }
+
+
+        /// Zero out off-diagonal blocks on rows corresponding to overlap cells
+        /// Diagonal blocks on ovelap rows are set to diag(1.0).
+        void makeOverlapRowsInvalid(Matrix& matrix) const
+        {
+            //value to set on diagonal
+            const int numEq = Matrix::block_type::rows;
+            typename Matrix::block_type diag_block(0.0);
+            for (int eq = 0; eq < numEq; ++eq)
+                diag_block[eq][eq] = 1.0;
+
+            //loop over precalculated overlap rows and columns
+            for (auto row = overlapRows_.begin(); row != overlapRows_.end(); row++ )
+                {
+                    int lcell = *row;
+                    // Zero out row.
+                    matrix[lcell] = 0.0;
+
+                    //diagonal block set to diag(1.0).
+                    matrix[lcell][lcell] = diag_block;
+                }
+        }
```

Comment:
There is `detail::makeOverlapRowsInvalid(Matrix, vector)`. Please simply use that...